src/java/packageTreeNode.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { INodeData, NodeKind } from "./nodeData";
+
+export class PackageTreeNode {
+    public name: string;
+    public fullName: string;
+    public childs: PackageTreeNode[] = [];
+    public isPackage: boolean = false;
+
+    constructor(packageName: string, parentName: string) {
+        const splitPackageName = packageName.split(".");
+        this.name = splitPackageName[0];
+        this.fullName = parentName === "" ? this.name : parentName + "." + this.name;
+        if (splitPackageName.length > 1) {
+            this.childs.push(new PackageTreeNode(packageName.substring(this.name.length + 1), this.fullName));
+        } else {
+            this.isPackage = true;
+        }
+    }
+
+    public addPackage(packageName: string): void {
```

Comment:
addPackage constructor  compressTree should be better organized to clearance to detect tree structures based on a list of packages:
a.b
a.b.c.d
a.b.c.d.e

Reply:
Second this. Currently, the compressTree will not work for this case.